src/nft/components/layout/Marquee.tsx

```diff
@@ -0,0 +1,59 @@
+import { Box } from 'nft/components/Box'
+import * as styles from 'nft/components/layout/Marquee.css'
+import { ReactNode, useEffect, useRef, useState } from 'react'
+
+export const Marquee = ({ children, speed = 20 }: { children: ReactNode; speed?: number }) => {
+  const [containerWidth, setContainerWidth] = useState(0)
```

Comment:
The widths do not need to be included in state - they are used immediately after being computed.